protocol/communities/manager.go

```diff
@@ -1455,6 +1482,130 @@ func (m *Manager) IsSeedingHistoryArchiveTorrent(communityID types.HexBytes) boo
 	return exists && ok
 }
 
+func (m *Manager) DownloadHistoryArchivesByMagnetlink(communityID types.HexBytes, magnetlink string) ([]string, error) {
+
+	id := communityID.String()
+	ml, _ := metainfo.ParseMagnetUri(magnetlink)
+
+	log.Println("Adding torrent from magnetlink for community: ", id)
+	torrent, err := m.torrentClient.AddMagnet(magnetlink)
+	if err != nil {
+		return nil, err
+	}
+	m.torrentTasks[id] = ml.InfoHash
+
+	<-torrent.GotInfo()
```

Comment:
So here we're waiting for the magnetlink to emit its metadata (this is needed for us to know what files the torrent will have, so having this info is crucial).

During testing, I've experienced that it sometimes takes a long time to receive a signal, at which point nothing happens here.

Reply:
What happens if you are offline? We should probably timeout this if it could hang forever